internal/services/search/search_service_resource.go

```diff
@@ -235,25 +463,50 @@ func resourceSearchServiceRead(d *pluginsdk.ResourceData, meta interface{}) erro
 		d.Set("sku", skuName)
 
 		if props := model.Properties; props != nil {
-			partitionCount := 0
-			replicaCount := 0
-			publicNetworkAccess := false
+			partitionCount := 1         // Default
+			replicaCount := 1           // Default
+			publicNetworkAccess := true // publicNetworkAccess defaults to true...
+			cmkEnforcement := false     // cmkEnforcment defaults to false...
+			hostingMode := services.HostingModeDefault
+			authenticationOptions := make([]interface{}, 0)
 
 			if count := props.PartitionCount; count != nil {
-				partitionCount = int(*count)
+				partitionCount = int(pointer.From(count))
 			}
 
 			if count := props.ReplicaCount; count != nil {
-				replicaCount = int(*count)
+				replicaCount = int(pointer.From(count))
 			}
 
+			// NOTE: There is a bug in the API where it returns the PublicNetworkAccess value
+			// as 'Disabled' instead of 'disabled'
 			if props.PublicNetworkAccess != nil {
-				publicNetworkAccess = *props.PublicNetworkAccess != "Disabled"
+				publicNetworkAccess = strings.EqualFold(string(pointer.From(props.PublicNetworkAccess)), string(services.PublicNetworkAccessEnabled))
+			}
+
+			if props.HostingMode != nil {
+				hostingMode = *props.HostingMode
 			}
 
+			var cmkCompliance string
+			if props.EncryptionWithCmk != nil {
+				cmkEnforcement = strings.EqualFold(string(pointer.From(props.EncryptionWithCmk.Enforcement)), string(services.SearchEncryptionWithCmkEnabled))
+				cmkCompliance = string(pointer.From(props.EncryptionWithCmk.EncryptionComplianceStatus))
+			}
```

Comment:
I am not sure, but I would lean toward exposing it rather than hide it as we don't know how the end-users would use this field. On a side note, what is the harm in having it exposed even if no one uses it?

Reply:
Fixed.